src/FSharpVSPowerTools.Logic/Navigation/GoToDefinitionFilter.fs

```diff
@@ -184,26 +187,23 @@ type GoToDefinitionFilter
         let uiContext = SynchronizationContext.Current
         let worker =
             async {
-                let! symbolResult = getDocumentState()
+                let getLexerState = vsLanguageService.GetQueryLexState textBuffer
+                let snapshotPoint = textBuffer.GetSnapshotPoint view.Caret.Position
+                let pointInDocument = vsLanguageService.MakePointInDocument snapshotPoint doc.FilePath
+                let! symbolResult = getDocumentState pointInDocument getLexerState
                 match symbolResult with
-                | Some (_, _, _, _, FSharpFindDeclResult.DeclFound _) 
+                | Some FoundInternal
                 | None ->
-                    // no FSharpSymbol found, here we look at #load directive
-                    let! directive = 
-                        asyncMaybe {
-                            let! project = project()
-                            return! vsLanguageService.GetLoadDirectiveFileNameAtCursor(doc.FilePath, view, project)
-                        }
-                    match directive with
-                    | Some fileToOpen ->
-                        // directive found, navigate to the file at first line
-                        serviceProvider.NavigateTo(fileToOpen, startRow=0, startCol=0, endRow=0, endCol=0)
-                    | None ->
-                        // Run the operation on UI thread since continueCommandChain may access UI components.
-                        do! Async.SwitchToContext uiContext
-                        // Declaration location might exist so let Visual F# Tools handle it. 
-                        return continueCommandChain()
-                | Some (project, parseTree, span, fsSymbolUse, FSharpFindDeclResult.DeclNotFound _) ->
+                    // Run the operation on UI thread since continueCommandChain may access UI components.
+                    do! Async.SwitchToContext uiContext
+                    // Declaration location might exist so let Visual F# Tools handle it. 
+                    return continueCommandChain()
+                | Some (FoundLoadDirective fileToOpen) -> 
+                    // directive found, navigate to the file at first line
+                    serviceProvider.NavigateTo(fileToOpen, startRow=0, startCol=0, endRow=0, endCol=0)
+                | Some (FoundExternal (project, parseTree, symbol, fsSymbolUse)) ->
+                    let snapshotPoint = snapshotPoint |> Option.get // this Option.get is guaranted safe because we actually had a match at position
```

Comment:
Logically you are right, the code is correct, but the correctness is not enforced by the compiler. It's not good.